addons/point_of_sale/static/src/js/screens.js

```diff
@@ -1182,7 +1182,7 @@ var ClientListScreenWidget = ScreenWidget.extend({
         
         var fields = {};
         this.$('.client-details-contents .detail').each(function(idx,el){
-            fields[el.name] = el.value;
+            fields[el.name] = el.value || false;
```

Comment:
The `false` was to match the `fields.id = partner.id || false;` just below. Not sure which is better. 
Do you mean something like
```javascript
if (el.value && partner.id)
    fields[el.name] = el.value;
```
? We can skip the field only for the create, not the update, as it would prevent to remove a value.